core/src/main/java/org/apache/calcite/sql/type/FunctionSqlType.java

```diff
@@ -58,6 +68,20 @@ public FunctionSqlType(
     return this;
   }
 
+  /**
+   * Returns the parameter type of the function.
+   *
+   * @return a struct wrapping function's parameter types.
+   */
+  public RelDataType getParameterType() {
```

Comment:
I think calling this `getParameterTypes` is more indicative of the fact that this may encode multiple parameters. Another choice is `getParameterSignatures`.

Reply:
nothing prevents us from changing the constructor parameter name and the field name - they are private to this class. I think anything that makes it less likely for users to make mistakes is a good change.